irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -0,0 +1,800 @@
+                                       q.accountId(),
+                                       Role::kGetMyAccount,
+                                       Role::kGetAllAccounts,
+                                       Role::kGetDomainAccounts))
+                     .str();
+      soci::statement st = (sql_.prepare << cmd);
+      st.exchange(soci::use(q.accountId(), "target_account_id"));
+      st.exchange(soci::into(tuple));
+
+      st.define_and_bind();
+      st.execute(true);
+
+      if (not tuple.get<5>()) {
+        return statefulFailed();
+      }
+
+      if (not tuple.get<0>().is_initialized()) {
+        return buildError<shared_model::interface::NoAccountErrorResponse>();
+      }
+
+      auto roles_str = tuple.get<4>().get();
+      roles_str.erase(0, 1);
+      roles_str.erase(roles_str.size() - 1, 1);
+
+      std::vector<shared_model::interface::types::RoleIdType> roles;
+
+      boost::split(roles, roles_str, [](char c) { return c == ','; });
+
+      auto account = fromResult(factory_->createAccount(q.accountId(),
+                                                        tuple.get<1>().get(),
+                                                        tuple.get<2>().get(),
+                                                        tuple.get<3>().get()));
+
+      if (not account) {
+        return statefulFailed();
+      }
+
+      auto response = QueryResponseBuilder().accountResponse(
+          *std::static_pointer_cast<shared_model::proto::Account>(*account),
+          roles);
+      return response;
+    }
+
+    QueryResponseBuilderDone PostgresQueryExecutorVisitor::operator()(
+        const shared_model::interface::GetSignatories &q) {
+      std::vector<shared_model::interface::types::PubkeyType> pubkeys;
+      soci::indicator ind;
+      boost::tuple<boost::optional<std::string>, int> row;
+      auto cmd = (boost::format(R"(WITH has_perms AS (%s),
+      t AS (
+          SELECT public_key FROM account_has_signatory
+          WHERE account_id = :account_id
+      )
+      SELECT public_key, perm FROM t
+      RIGHT OUTER JOIN has_perms ON TRUE
+      )")
+                  % hasQueryPermission(creator_id_,
+                                       q.accountId(),
+                                       Role::kGetMySignatories,
+                                       Role::kGetAllSignatories,
+                                       Role::kGetDomainSignatories))
+                     .str();
+      soci::statement st = (sql_.prepare << cmd);
+      st.exchange(soci::use(q.accountId()));
+      st.exchange(soci::into(row, ind));
+
+      st.define_and_bind();
+      st.execute();
+
+      int has_perm = -1;
+
+      processSoci(st,
+                  ind,
+                  row,
+                  [&pubkeys, &has_perm](
+                      boost::tuple<boost::optional<std::string>, int> &row) {
+                    has_perm = row.get<1>();
+                    if (row.get<0>()) {
```

Comment:
Also please check where bind operator (`operator|`) is applicable (e.g. where else branch is not required) to increase readability.